### PR TITLE
Allow items to be expanded initially

### LINKED ISSUE #36: Option to have parents open on load
It would be great to be able to define one (or more) parents to be expanded when the adapter first loads. Great library btw thanks

### PR DESCRIPTION
**Pull request by paul-turner**
Introduces a new method that allows for a Parent Object to initially show up as expanded.
Also cleaned up some casts and duplicated lists in the expanded adapter.

@rbro112 if you have time would like your opinion on some of the changes in ExpandableRecyclerAdapter

Fixes #36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,7 +52,37 @@ Javadocs for the library and sample are available [here](http://bignerdranch.git
  
  Then, create two ViewHolders and their respective layouts. One ViewHolder must extend ```ParentViewHolder``` and the other must extend ```ChildViewHolder```. Create their respective views and create variables in these ViewHolders to access those views.
  
- Next, the Object that contains the data to be displayed in your RecyclerView must extend ```ParentObject```
+ Next, the Object that contains the data to be displayed in your RecyclerView must extend ```ParentObject```. When you implement ```ParentObject``` there are three methods that you must implement, shown in the example below:
```
Comment (alunsford3) [inline README.md]: I may be late to the game, but why are these required? Why are they not optional, and allowing the superclass to do these actions by default?

Comment (paul-turner) [inline README.md]: No worries: We went with an interface to allow for whatever object hierarchy you want while implementing `ParentObject` interface, instead of having to extend out class.

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 2]
```diff
@@ -169,12 +165,12 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
...
-            parentViewHolder.setExpanded(((ParentWrapper) mExpandableRecyclerAdapterHelper.getHelperItemAtPosition(position)).isExpanded());
-            onBindParentViewHolder(parentViewHolder, position, mItemList.get(position));
-        } else if (mItemList.get(position) == null) {
+            parentViewHolder.setExpanded(((ParentWrapper) getHelperItem(position)).isExpanded());
+            onBindParentViewHolder(parentViewHolder, position, ((ParentWrapper) getHelperItem(position)).getParentObject());
+        } else if (getHelperItem(position) == null) {
             throw new IllegalStateException("Incorrect ViewHolder found");
         } else {
-            onBindChildViewHolder((CVH) holder, position, mItemList.get(position));
+            onBindChildViewHolder((CVH) holder, position, getHelperItem(position));
```
Comment (alunsford3) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Could we maybe not call `getHelperItem` so many times?

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 3]
```diff
@@ -437,20 +435,24 @@ public void onRestoreInstanceState(Bundle savedInstanceStateBundle) {
         }
         mStableIdMap = (HashMap<Long, Boolean>) savedInstanceStateBundle.getSerializable(STABLE_ID_MAP);
         int i = 0;
-        while (i < mExpandableRecyclerAdapterHelper.getHelperItemList().size()) {
-            if (mExpandableRecyclerAdapterHelper.getHelperItemAtPosition(i) instanceof ParentWrapper) {
-                ParentWrapper parentWrapper = (ParentWrapper) mExpandableRecyclerAdapterHelper.getHelperItemAtPosition(i);
+        while (i < mHelperItemList.size()) {
+            if (getHelperItem(i) instanceof ParentWrapper) {
```
Comment (alunsford3) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Could we keep using `Object helperItem = getHelperItem(i);` to reduce calls to `getHelperItem`?

**Comment by paul-turner** [comment]
reduced helper item calls

**Comment by rbro112** [comment]
@paul-turner Pretty busy with classes but I'll be taking a look later on tonight. Thanks for fixing it up.

File: app/src/main/java/com/ryanbrooks/expandablerecyclerviewsample/VerticalLinearRecyclerViewSample/CustomParentObject.java [hunk 4]
```diff
@@ -20,6 +20,7 @@
 
     private String mParentText;
     private int mParentNumber;
+    private boolean mInitiallyExpand;
```
Comment (dgreenhalgh) [inline app/src/main/java/com/ryanbrooks/expandablerecyclerviewsample/VerticalLinearRecyclerViewSample/CustomParentObject.java]: This may be a bit picky, but the proper name of this variable according to our naming scheme would be `mInitiallyExpanded`. The getter below is correct, but you'd need to rename the setter as well.
